### PR TITLE
Implement asyncArray

### PR DESCRIPTION
**Pull request by bugagashenkj**
#385

### CODE REVIEW

File: lib/async.array.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  [Symbol.asyncIterator]() {
+    let sum = 0;
+    let count = 0;
+    let begin = Date.now();
+    let iterMax = this.min;
+
+    const getItem = () => {
+      const item = { value: this[count], done: count === this.length };
```
Comment (nechaido) [inline lib/async.array.js]:
```suggestion
      const item = { value: this[count], done: count >= this.length };
```
AsyncIterator interface defines that any consecutive calls to `next` after consuming an iterrators should return:
```js
{ value: undefined, done: false }
```

File: .eslintrc.yml [hunk 2]
```diff
@@ -1 +1,2 @@
 extends: 'metarhia'
+parserOptions: { "ecmaVersion": 2019 }
```
Comment (aqrln) [inline .eslintrc.yml]: We should probably add this upstream

File: lib/async.array.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    this.min = 5;
+    this.ratio = 1.5;
+  }
+
+  set percent(value) {
+    if (value < 1 && value >= 0) {
+      this.ratio = value / (1 - value);
+    } else {
+      throw new Error('The percen must be less than 1 and greater ' +
```
Comment (nechaido) [inline lib/async.array.js]:
```suggestion
      throw new Error('The percent must be less than 1 and greater ' +
```

Comment (nechaido) [reply lib/async.array.js]: This should be RangeError, also I'd use `...must be >= 0 and < 1` or even `...must be in range [0, 1)`.

File: lib/async.array.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  [Symbol.asyncIterator]() {
+    let sum = 0;
+    let count = 0;
+    let begin = Date.now();
+    let iterMax = this.min;
+
+    const getItem = () => {
+      const done = count >= this.length;
+      const value = done ? undefined : this[count];
```
Comment (nechaido) [inline lib/async.array.js]: You don't need this check as it is already performed by an array, though no strong feelings about this one.

**Pull request closed without merging**